teos/gatekeeper.py

```diff
@@ -0,0 +1,117 @@
+from common.tools import is_compressed_pk
+from common.cryptographer import Cryptographer
+
+
+class NotEnoughSlots(ValueError):
+    """Raise this when trying to subtract more slots than a user has available"""
```

Comment:
I'd write the descriptions as "Raised when...", so it's equally good for the programmer raising it and for the one that might want to catch it.